sqld/src/connection/write_proxy.rs

```diff
@@ -293,6 +293,14 @@ impl Connection for WriteProxyConnection {
         self.wait_replication_sync().await?;
         self.read_db.describe(sql, auth).await
     }
+
+    async fn is_autocommit(&self) -> Result<bool> {
+        let state = self.state.lock().await;
+        Ok(match *state {
+            State::Txn => false,
+            State::Init | State::Invalid => true,
```

Comment:
I was hoping that the invalid state would not occur in practice :) Assuming autocommit is safer, because it is more dangerous to execute a statement that is intended to be in transaction outside of a transaction. (I.e., it is better if the client aborts a transaction if the server reaches the invalid state.)